datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -322,7 +322,7 @@ def _normalize_tags(self, tags, device_name):
 
         if device_name:
             self._log_deprecation("device_name")
-            normalized_tags.append("device:%s" % device_name)
+            normalized_tags.append("device:{}".format(device_name))
```

Comment:
According to the docstring of the method, the `device` tag should be appended to a copy of tags before the call to `_normalize_tags_type`.